scripts/templates/README.md

```diff
@@ -0,0 +1,163 @@
+* aims to support the basic and usual "kinds" of detectors (threshold, heartbeat..).
+* for less common implementation it will require additional manual updates.
+
+This is an helper, it should not be fully trusted and could be not up to date:
+
+* focus on monitoring, generate code, push it, if CI pass so deploy detector and test it 
+with real data. It could have errors from your config or the templates, tests remain crucial.
+* it does not aim to replace terraform "templating". we still ideally want to create a 
+common submodule used by every detectors modules (see [this 
+issue](https://github.com/claranet/terraform-signalfx-detectors/issues/105).
+* it is complementary waiting to implement this issue or to update existing base to use it
+when it will be available. It still could be useful to decorrelate some policies from the
+"deliverable" and compensate for terraform lacks like [this 
+feature](https://github.com/hashicorp/terraform/issues/19898).
+
+### Requirements
+
+* Jinja2 compatible renderer. 
+Next sections will use [j2cli](https://github.com/kolypto/j2cli#installation).
+
+### Structure
+
+At the root of this directory there are:
+
+* [values.yaml](./values.yaml) with all configuration options available.
+Copy it to create your own preset config.
+* jinja templates files `*.j2` like [the detector resource](./detector-res.tf.j2).
+* Sub-directories containing preset configs to generate common detectors on this repo.
+
+The templates files are splitted into different files depending on the destination
+where to append code:
+
+* `variables.tf` file from [./detector-vars.tf.j2](./detector-vars.tf.j2).
+* `detectors-{mymodule}.tf` file from [./detector-res.tf.j2](./detector-res.tf.j2).
+
+### Usage
+
+Use `j2cli` with template file and your config file created from [values.yaml](./values.yaml):
+
+```bash
+$ j2 ./scripts/templates/detector-{template}.tf.j2 {path_to_config_file.yaml}
+```
+
+You can redirect the output in the destination file on the repo:
+
+```bash
+$ j2 ./scripts/templates/detector-vars.tf.j2 ./scripts/templates/threshold/nginx.yaml >> ./middleware/nginx/variables.tf
+```
+
+__Note__: you can duplicate another existing module and empty `variables.tf` and 
+`detectors-{module}.tf` files if it does not exist yet.
+
+### Examples
+
+To create a new detector use each required templates:
+
+```bash
+$ j2 ./scripts/templates/detector_res.tf.j2 scripts/templates/threshold/simple.yaml
+```
+
+this will generate [detector terraform 
+resource](https://registry.terraform.io/providers/splunk-terraform/signalfx/latest/docs/resources/detector):
+
+```hcl
+resource "signalfx_detector" "real_time_requests_latency" {
+  name = format("%s %s", local.detector_name_prefix, "Database Real time requests latency")
+
+  program_text = <<-EOF
+    signal = data('read_latency_ns_p99', filter=${module.filter-tags.filter_custom})${var.real_time_requests_latency_transformation_function}
```

Comment:
we don't need .publish('signal') at the end?

Reply:
already fixed (commit not pushed yet)

Reply:
seems not fixed?

Reply:
you are totally right. fixed in

Reply:
I don't manage to reproduce, seems fixed for me ^^